File: test_complete_flow.mjs
// Test script to verify the complete flow from XSchedule-X
async function makeRequest(url, options = {}) {
  try {
    const response = await fetch(url, options);
    const text = await response.text();
    try {
      return JSON.parse(text);
    } catch {
      return { success: false, error: text };
    }
  } catch (error) {
    return { success: false, error: error.message };
  }
}

async function testCompleteFlow() {
  console.log('🧪 Testing complete flow from XSchedule-X...\n');
  
  try {
    // Step 1: Create a simple test prompt via API
    console.log('1️⃣ Creating test prompt via API...');
    const promptData = {
      name: 'Simple Test Prompt',
      content: 'Generate a tweet that ends with the word SUCCESS in capital letters.',
      variables: ['topic'],
      is_active: true
    };
    
    const saveResponse = await makeRequest('http://localhost:3000/api/config/prompts', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(promptData)
    });
    
    if (!saveResponse.success) {
      console.log('❌ Failed to save prompt:', saveResponse.error);
      return;
    }
    
    console.log('✅ Prompt saved:', saveResponse.prompt.name);
    
    // Step 2: Verify the current prompt is set
    console.log('\n2️⃣ Verifying current prompt...');
    const currentResponse = await makeRequest('http://localhost:3000/api/config/prompts/current');
    
    if (!currentResponse.success) {
      console.log('❌ Failed to get current prompt:', currentResponse.error);
      return;
    }
    
    console.log('✅ Current prompt:', currentResponse.prompt?.name || 'No prompt set');
    console.log('   Content:', currentResponse.prompt?.content || 'No content');
    
    // Step 3: Test generation like the frontend does
    console.log('\n3️⃣ Testing generation (frontend style)...');
    
    // This mimics how the frontend calls generateAI()
    const generateResponse = await makeRequest('http://localhost:3000/api/generate', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({
        mode: 'ops',
        prompt: currentResponse.prompt.content,
        promptId: currentResponse.prompt.id,
        promptVariables: currentResponse.prompt.variables || []
      })
    });
    
    if (!generateResponse.success) {
      console.log('❌ Generation failed:', generateResponse.error);
      return;
    }
    
    console.log('✅ Generation successful!');
    const text = generateResponse.variants[0]?.text || '';
    console.log('   Generated text:', text);
    
    // Check if it ends with SUCCESS
    const endsWithSuccess = text.trim().toUpperCase().endsWith('SUCCESS');
    console.log(`   ${endsWithSuccess ? '✅' : '❌'} Ends with SUCCESS: ${endsWithSuccess}`);
    
    // Step 4: Test the actual frontend endpoint
    console.log('\n4️⃣ Testing actual frontend generation...');
    
    // This is exactly how the frontend calls it
    const frontendResponse = await makeRequest('http://localhost:3000/api/generate', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ mode: 'ops' }) // Frontend doesn't send prompt, it gets it from current
    });
    
    if (!frontendResponse.success) {
      console.log('❌ Frontend generation failed:', frontendResponse.error);
      return;
    }
    
    console.log('✅ Frontend generation successful!');
    const frontendText = frontendResponse.variants[0]?.text || '';
    console.log('   Generated text:', frontendText);
    
    // Check if it ends with SUCCESS
    const frontendEndsWithSuccess = frontendText.trim().toUpperCase().endsWith('SUCCESS');
    console.log(`   ${frontendEndsWithSuccess ? '✅' : '❌'} Ends with SUCCESS: ${frontendEndsWithSuccess}`);
    
    console.log('\n🎉 Complete flow test finished!');
    console.log('   The prompt injection system is working correctly.');
    console.log('   The LLM may not always follow specific instructions like ending with SUCCESS.');
    
  } catch (error) {
    console.log('❌ Test failed:', error.message);
  }
}

// Run the test
testCompleteFlow();

File: test_complete_flow.py
#!/usr/bin/env python3
"""
Test script to verify the complete prompt injection flow works end-to-end.
"""

import requests
import json

def test_complete_flow():
    print("🧪 Testing Complete Prompt Injection Flow")
    print("=" * 50)
    
    # Step 1: Check current prompt
    print("\n1️⃣ Checking current prompt...")
    try:
        response = requests.get("http://localhost:3000/api/config/prompts/current")
        prompt_data = response.json()
        print(f"✅ Current prompt endpoint working: {prompt_data.get('success')}")
        
        if prompt_data.get('success') and prompt_data.get('prompt'):
            current_prompt = prompt_data['prompt']['content']
            print(f"📄 Current prompt: {current_prompt}")
            print(f"🔍 Contains professional content: {'social media' in current_prompt.lower() or 'business' in current_prompt.lower()}")
        else:
            print("❌ No current prompt found")
            return False
            
    except Exception as e:
        print(f"❌ Error checking current prompt: {e}")
        return False
    
    # Step 2: Test generation through XSchedule-X proxy
    print("\n2️⃣ Testing generation through XSchedule-X...")
    try:
        response = requests.post(
            "http://localhost:3000/api/generate",
            json={"mode": "ops"},
            headers={"Content-Type": "application/json"}
        )
        
        if response.status_code == 200:
            data = response.json()
            print(f"✅ Generation successful, received {len(data.get('variants', []))} variants")
            
            # Check if quality content appears in any variant
            variants = data.get('variants', [])
            quality_found = False
            for i, variant in enumerate(variants):
                text = variant.get('text', '')
                print(f"Variant {i+1}: {text}")
                # Check for quality indicators
                if len(text) > 20 and any(word in text.lower() for word in ['business', 'technology', 'professional', 'insights']):
                    quality_found = True
                    print(f"✅ Quality content found in variant {i+1}!")
            
            if not quality_found:
                print("❌ No quality content found in any variant")
                return False
                
        else:
            print(f"❌ Generation failed: {response.status_code} - {response.text}")
            return False
            
    except Exception as e:
        print(f"❌ Error during generation: {e}")
        return False
    
    print("\n🎉 SUCCESS: Complete prompt injection flow working!")
    return True

if __name__ == "__main__":
    success = test_complete_flow()
    exit(0 if success else 1)